src/wavm/openmp.cpp

```diff
@@ -473,73 +470,81 @@ WAVM_DEFINE_INTRINSIC_FUNCTION(env,
         }
 
         logger->debug("Distributed Fork finished successfully");
-    } else { // Single host
+
+    } else if (conf.threadMode == "local") {
```

Comment:
This block has changed the most in this file. It used to use the thread pool and hence introduced a couple of layers of indirection. Now it just spawns threads using `std::thread`, so should be more self-contained. This is important as this section is likely to change a lot when we change the implementation.